### PR TITLE
Fix protobuf version dependency

### PR DESCRIPTION
**Pull request by sribalakumar**
Some of the compiled *_pb.rb files included for OTLP exporter is having `add_file` method and its definition is available only from [google-protobuf >= 3.7.0](https://github.com/protocolbuffers/protobuf/blob/v3.7.0/ruby/ext/google/protobuf_c/defs.c#L2124)
```ruby
  Google::Protobuf::DescriptorPool.generated_pool.build do
    add_file("{path_to_proto_file}", :syntax => :proto3) do
      ...
    end
  end
  ```
  
  `add_file` is currently used in the below files:
- https://github.com/open-telemetry/opentelemetry-ruby/blob/master/exporter/otlp/lib/opentelemetry/proto/collector/trace/v1/trace_service_pb.rb#L8
- https://github.com/open-telemetry/opentelemetry-ruby/blob/master/exporter/otlp/lib/opentelemetry/proto/common/v1/common_pb.rb#L7
- https://github.com/open-telemetry/opentelemetry-ruby/blob/master/exporter/otlp/lib/opentelemetry/proto/resource/v1/resource_pb.rb
- https://github.com/open-telemetry/opentelemetry-ruby/blob/master/exporter/otlp/lib/opentelemetry/proto/trace/v1/trace_pb.rb#L9

### CODE REVIEW

**Review by fbogsany — APPROVED** [review]
Thanks @sribalakumar. Any reason not to merge this? (Not sure why it is still in Draft.)

**Comment by johnnyshields** [comment]
What is the reason for this?

**Comment by sribalakumar** [comment]
@johnnyshields Updated the PR description with the reason for this change. 

@fbogsany kept it in draft just to verify change tests and to update the PR description with the correct references.

File: exporter/otlp/opentelemetry-exporter-otlp.gemspec [hunk 1]
```diff
@@ -25,7 +25,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ['lib']
   spec.required_ruby_version = '>= 2.5.0'
 
-  spec.add_dependency 'google-protobuf', '>= 3.4.1.1', '< 4'
+  spec.add_dependency 'google-protobuf', '>= 3.7.0', '< 4'
```
Comment (dazuma) [inline exporter/otlp/opentelemetry-exporter-otlp.gemspec:28]: If we don't require a specific patch level, I'd just use

```ruby
spec.add_dependency 'google-protobuf', '~> 3.7'
```

**Review by dazuma — CHANGES_REQUESTED** [review]

File: exporter/otlp/opentelemetry-exporter-otlp.gemspec (hunk 1 above)
Comment (sribalakumar) [reply exporter/otlp/opentelemetry-exporter-otlp.gemspec]: Thanks, made the change to refer just by minor version and ignore patch level as you suggested.

**Review by dazuma — APPROVED** [review]

**Pull request merged**